File: client/src/game/conf/PartConfig.ts
class PartConfig {
    private static _dic:Dictionary<PartConfig> = new Dictionary<PartConfig>();
    private static _arr:Array<PartConfig> = new Array<PartConfig>();

	 public Id :number ;
     public ArmName : Array<string>  =new Array<string>();
    
     public Type :number ;
     public TexName : Array<string>  =new Array<string>();
     public Slots : Array<number>  =new Array<number>();
     public MeSlots : Array<number>  =new Array<number>();
     public Show :number ;
     public IconScale :number ;
     public Deviation : Array<number>  =new Array<number>();
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<PartConfig>{
        return  PartConfig._dic;
    }
    public static get Arr():Array<PartConfig>{
        return PartConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:PartConfig = <PartConfig>json[i];
				PartConfig._arr.push(config);
				PartConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("PartConfig loader fail!");
        }
    }
}

File: server/src/game/conf/PartConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type PartConfigElem struct {
	Id int
	ArmName []string
	Name string
	Type int
	TexName []string
	Slots []int
	MeSlots []int
	Show int
	IconScale float32
	Deviation []float32
	
}

type PartConfig0 struct {
	Dic map[int]*PartConfigElem
	Arr []*PartConfigElem
}

var (
	PartConfig PartConfig0
)

func loadPartConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	PartConfig.Dic = make(map[int]*PartConfigElem)
	PartConfig.Arr = make([]*PartConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &PartConfig.Arr)
	for _,conf := range PartConfig.Arr{
		PartConfig.Dic[conf.Id] = conf
	}
	
}
